graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/ESPivotBucketSpecHandler.java

```diff
@@ -77,7 +92,19 @@ protected List<BucketOrder> orderListForPivot(Pivot pivot, ESGeneratedQueryConte
                 })
                 .filter(Objects::nonNull)
                 .collect(Collectors.toList());
-        return ordering.isEmpty() ? List.of(defaultOrder) : ordering;
+        return ordering.isEmpty()
+                ? new SortOrders(List.of(defaultOrder), List.of())
+                : new SortOrders(ordering, List.copyOf(subaggregations));
+    }
+
+    private boolean isSortOnNumericPivotField(Pivot pivot, PivotSort pivotSort, ESGeneratedQueryContext queryContext, Query query) {
+        return queryContext.fieldType(query.effectiveStreams(pivot), pivotSort.field())
+                .filter(this::isNumericFieldType)
+                .isPresent();
+    }
+
+    private boolean isNumericFieldType(String fieldType) {
+        return fieldType.equals("long") || fieldType.equals("double") || fieldType.equals("float");
     }
```

Comment:
The fact that we duplicate this method may be also an indicator that we should rather move a code, that is now spread in two classes, to a single class.